Build failure: linuxPackages_latest.yt6801

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

Use the yt6801 driver in `boot.extraModulePackages`, such as on the [TUXEDO InfinityBook Gen 9](https://github.com/NixOS/nixos-hardware/blob/master/tuxedo/infinitybook/pro14/gen9/default.nix)

### Can Hydra reproduce this build failure?

Please select the Hydra Status.

### Link to Hydra build job

_No response_

### Relevant log output

```console
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/6qsd1z428mwpjv6dic3fcngjq09a4c8r-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/40dn8zz0viwpz5q25jn0iizra1nwp0i7-kernel_6.15_fix.patch
patching file src/fuxi-gmac-net.c
Hunk #1 succeeded at 784 (offset -86 lines).
patching file src/fuxi-gmac-phy.c
Hunk #1 succeeded at 490 (offset 122 lines).
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: SHELL=/nix/store/gkwbw9nzbkbz298njbn3577zmrnglbbi-bash-5.3p0/bin/bash KERNELRELEASE=6.16.0 KSRC_BASE= KSRC=/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/build KDST=kernel/drivers/net/ethernet/motorcomm INSTALL_MOD_PATH=/nix/store/y4w8fhy8r0pn1is82vdsp7ill0ifh5a2-yt6801-1.0.29-20240812 ko_dir=\$\(out\)/lib/modules/6.16.0/kernel/drivers/net/ethernet/motorcomm ko_full=\$\(out\)/lib/modules/6.16.0/kernel/drivers/net/ethernet/motorcomm/yt6801.ko.xz

CC:  gcc
CCVERSION:  14.3.0
KERNEL_GCC_VERSION: 
KVER:  6.16.0


make -C src/ clean
make[1]: Entering directory '/build/source/src'
make -C /nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/build M=/build/source/src clean
make[2]: Entering directory '/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/build'
make[3]: Entering directory '/build/source/src'
make[3]: Leaving directory '/build/source/src'
make[2]: Leaving directory '/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/build'
make[1]: Leaving directory '/build/source/src'


make -C src/ modules
make[1]: Entering directory '/build/source/src'
make -C /nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/build M=/build/source/src modules
make[2]: Entering directory '/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/build'
make[3]: Entering directory '/build/source/src'
  CC [M]  fuxi-gmac-common.o
fuxi-gmac-common.c: In function 'fxgmac_drv_probe':
fuxi-gmac-common.c:487:64: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
  487 |                                     netdev->real_num_tx_queues);
      |                                                                ^
fuxi-gmac-common.c: In function 'fxgmac_get_all_hw_features':
fuxi-gmac-common.c:630:52: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
  630 |         DPRINTK ("Mac ver=%#x\n", hw_feat->version);
      |                                                    ^
  CC [M]  fuxi-gmac-desc.o
fuxi-gmac-desc.c: In function 'fxgmac_alloc_rings':
fuxi-gmac-desc.c:242:78: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
  242 |                             DPRINTK(" tx_desc_cnt=%u,", pdata->tx_desc_count);
      |                                                                              ^
fuxi-gmac-desc.c: In function 'fxgmac_free_channels':
fuxi-gmac-desc.c:265:95: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
  265 |     if(netif_msg_drv(pdata)) DPRINTK("free_channels,tx_ring=%p", pdata->channel_head->tx_ring);
      |                                                                                               ^
fuxi-gmac-desc.c:269:83: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
  269 |     if(netif_msg_drv(pdata)) DPRINTK(" ,rx_ring=%p", pdata->channel_head->rx_ring);
      |                                                                                   ^
fuxi-gmac-desc.c:273:76: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
  273 |     if(netif_msg_drv(pdata)) DPRINTK(" ,channel=%p\n", pdata->channel_head);
      |                                                                            ^
fuxi-gmac-desc.c: In function 'fxgmac_alloc_channels':
fuxi-gmac-desc.c:417:155: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
  417 |     if(netif_msg_drv(pdata)) DPRINTK("alloc_channels,channel_head=%p,size=%d*%d\n", channel_head, pdata->channel_count,(u32)sizeof(struct fxgmac_channel));
      |                                                                                                                                                           ^
fuxi-gmac-desc.c:427:142: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
  427 |     if(netif_msg_drv(pdata)) DPRINTK("alloc_channels,tx_ring=%p,size=%d*%d\n", tx_ring, pdata->tx_ring_count,(u32)sizeof(struct fxgmac_ring));
      |                                                                                                                                              ^
fuxi-gmac-desc.c:433:142: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
  433 |     if(netif_msg_drv(pdata)) DPRINTK("alloc_channels,rx_ring=%p,size=%d*%d\n", rx_ring, pdata->rx_ring_count,(u32)sizeof(struct fxgmac_ring));
      |                                                                                                                                              ^
fuxi-gmac-desc.c:485:68: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
  485 |     if(netif_msg_drv(pdata)) DPRINTK("alloc_channels callout ok\n");
      |                                                                    ^
  CC [M]  fuxi-gmac-ethtool.o
fuxi-gmac-ethtool.c: In function 'fxgmac_set_rss_hash_opt':
fuxi-gmac-ethtool.c:641:72: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
  641 |                        " may arrive out of order to the stack above\n");
      |                                                                        ^
  CC [M]  fuxi-gmac-hw.o
fuxi-gmac-hw.c: In function 'fxgmac_config_rss':
fuxi-gmac-hw.c:2726:62: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
 2726 |         DPRINTK("fxgmac_config_rss: error configuring RSS\n");
      |                                                              ^
  CC [M]  fuxi-gmac-net.o
fuxi-gmac-net.c: In function 'fxgmac_get_netdev_ip6addr':
fuxi-gmac-net.c:243:89: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
  243 |         DPRINTK ("%s FXGMAC_NS_IFA_GLOBAL_UNICAST is set, %x\n", __FUNCTION__, ifa_flag);
      |                                                                                         ^
fuxi-gmac-net.c:246:85: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
  246 |         DPRINTK ("%s FXGMAC_NS_IFA_LOCAL_LINK is set, %x\n", __FUNCTION__, ifa_flag);
      |                                                                                     ^
fuxi-gmac-net.c:271:77: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
  271 |     if(err) DPRINTK("%s get ipv6 addr failed, use default.\n", __FUNCTION__);
      |                                                                             ^
fuxi-gmac-net.c: In function 'fxgmac_maybe_stop_tx_queue':
fuxi-gmac-net.c:343:80: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
  343 |             if(netif_msg_tx_done(pdata)) DPRINTK("about stop tx q, ret BUSY\n");
      |                                                                                ^
fuxi-gmac-net.c: In function 'fxgmac_prep_tx_pkt':
fuxi-gmac-net.c:421:140: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
  421 |         DPRINTK ("fxgmac_prep_tx_pkt callin,pkt desc cnt=%d,skb len=%d, skbheadlen=%d\n", pkt_info->desc_count, skb->len, skb_headlen(skb));
      |                                                                                                                                            ^
fuxi-gmac-net.c:430:168: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
  430 |             DPRINTK("fxgmac_is_tso=%d, ip_summed=%d,skb gso=%d\n",((skb->ip_summed == CHECKSUM_PARTIAL) && (skb_is_gso(skb)))?1:0, skb->ip_summed, skb_is_gso(skb)?1:0);
      |                                                                                                                                                                        ^
fuxi-gmac-net.c:445:113: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
  445 |         if(netif_msg_tx_done(pdata)) DPRINTK ("fxgmac_prep_tx_pkt,tso, pkt desc cnt=%d\n", pkt_info->desc_count);
      |                                                                                                                 ^
fuxi-gmac-net.c:467:149: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
  467 |         if(netif_msg_tx_done(pdata)) DPRINTK ("fxgmac_prep_tx_pkt,VLAN, pkt desc cnt=%d,vlan=0x%04x\n", pkt_info->desc_count, skb_vlan_tag_get(skb));
      |                                                                                                                                                     ^
fuxi-gmac-net.c:483:177: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
  483 |         DPRINTK ("fxgmac_prep_tx_pkt callout,pkt desc cnt=%d,skb len=%d, skbheadlen=%d,frags=%d\n", pkt_info->desc_count, skb->len, skb_headlen(skb), skb_shinfo(skb)->nr_frags);
      |                                                                                                                                                                                 ^
fuxi-gmac-net.c: In function 'fxgmac_napi_enable':
fuxi-gmac-net.c:852:106: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
  852 |             if(netif_msg_drv(pdata)) DPRINTK("napi_enable, msix ch%d napi enabled done,add=%d\n", i, add);
      |                                                                                                          ^
fuxi-gmac-net.c: In function 'fxgmac_napi_disable':
fuxi-gmac-net.c:939:112: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
  939 |                 if(netif_msg_drv(pdata)) DPRINTK("napi_disable, msix ch%d napi disabled done,del=%d\n", i, del);
      |                                                                                                                ^
fuxi-gmac-net.c: In function 'fxgmac_request_irqs':
fuxi-gmac-net.c:1048:76: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
 1048 |                                         channel->expansion.dma_irq_name_tx);
      |                                                                            ^
fuxi-gmac-net.c:1095:81: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
 1095 |          i, (pdata->channel_head)[0].dma_irq,(pdata->channel_head)[i-1].dma_irq);
      |                                                                                 ^
fuxi-gmac-net.c: In function 'fxgmac_free_irqs':
fuxi-gmac-net.c:1183:105: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
 1183 |                 if(netif_msg_drv(pdata)) DPRINTK("fxgmac_free_irqs, MSIx irq_tx clear done, ch=%d\n", i);
      |                                                                                                         ^
fuxi-gmac-net.c:1205:96: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
 1205 |     if(netif_msg_drv(pdata)) DPRINTK("fxgmac_free_irqs, MSIx rx irq clear done, total=%d\n", i);
      |                                                                                                ^
fuxi-gmac-net.c: In function 'fxgmac_start':
fuxi-gmac-net.c:1322:68: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
 1322 |     if(netif_msg_drv(pdata)) DPRINTK("fxgmac start callin here.\n");
      |                                                                    ^
fuxi-gmac-net.c: In function 'fxgmac_net_powerup':
fuxi-gmac-net.c:1513:68: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
 1513 |     if(netif_msg_drv(pdata)) DPRINTK("fxgmac_net_powerup callin\n");
      |                                                                    ^
fuxi-gmac-net.c: In function 'fxgmac_net_powerdown':
fuxi-gmac-net.c:1540:76: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
 1540 |     if(netif_msg_drv(pdata)) DPRINTK("fxgmac_net_powerdown callin here.\n");
      |                                                                            ^
fuxi-gmac-net.c:1546:91: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
 1546 |     if(netif_msg_drv(pdata)) DPRINTK("fxgmac_net_powerdown continue with down process.\n");
      |                                                                                           ^
fuxi-gmac-net.c:1589:117: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
 1589 |     if(netif_msg_drv(pdata)) DPRINTK("fxgmac_net_powerdown callout, powerstate=%ld.\n", pdata->expansion.powerstate);
      |                                                                                                                     ^
fuxi-gmac-net.c: In function 'fxgmac_open':
fuxi-gmac-net.c:1598:61: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
 1598 |     if(netif_msg_drv(pdata)) DPRINTK("fxgmac_open callin\n");
      |                                                             ^
fuxi-gmac-net.c:1625:62: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
 1625 |     if(netif_msg_drv(pdata)) DPRINTK("fxgmac_open callout\n");
      |                                                              ^
fuxi-gmac-net.c: In function 'fxgmac_close':
fuxi-gmac-net.c:1644:62: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
 1644 |     if(netif_msg_drv(pdata)) DPRINTK("fxgmac_close callin\n");
      |                                                              ^
fuxi-gmac-net.c:1660:63: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
 1660 |     if(netif_msg_drv(pdata)) DPRINTK("fxgmac_close callout\n");
      |                                                               ^
fuxi-gmac-net.c: In function 'fxgmac_xmit':
fuxi-gmac-net.c:1696:106: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
 1696 |     if(netif_msg_tx_done(pdata)) DPRINTK("xmit callin, skb->len=%d,q=%d\n", skb->len, skb->queue_mapping);
      |                                                                                                          ^
fuxi-gmac-net.c:1737:102: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
 1737 |     if(netif_msg_tx_done(pdata)) DPRINTK("xmit,before hw_xmit, byte len=%d\n", tx_pkt_info->tx_bytes);
      |                                                                                                      ^
fuxi-gmac-net.c: In function 'fxgmac_tx_poll':
fuxi-gmac-net.c:2227:149: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
 2227 |         if(netif_msg_tx_done(pdata) && (channel->queue_index < pdata->tx_q_count)) DPRINTK("tx_poll, null point to ring %d\n", channel->queue_index);
      |                                                                                                                                                     ^
fuxi-gmac-net.c:2231:117: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
 2231 |         DPRINTK("tx_poll callin, ring_cur=%d,ring_dirty=%d,qIdx=%d\n", ring->cur, ring->dirty, channel->queue_index);
      |                                                                                                                     ^
fuxi-gmac-net.c: In function 'fxgmac_dev_xmit':
fuxi-gmac-net.c:2720:86: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
 2720 |     if(netif_msg_tx_done(pdata)) DPRINTK("dev_xmit callin, desc cur=%d\n", cur_index);
      |                                                                                      ^
fuxi-gmac-net.c:2749:173: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
 2749 |     if(vlan && (netif_msg_tx_done(pdata))) DPRINTK("fxgmac_dev_xmi:pkt vlan=%d, ring vlan=%d, vlan_context=%d\n", pkt_info->vlan_ctag, ring->tx.cur_vlan_ctag, vlan_context);
      |                                                                                                                                                                             ^
fuxi-gmac-net.c:2763:33: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
 2763 |                   pkt_info->mss);
      |                                 ^
fuxi-gmac-net.c:2919:100: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
 2919 |         cur_index, start_index, dma_desc->desc0, dma_desc->desc1, dma_desc->desc2, dma_desc->desc3);
      |                                                                                                    ^
fuxi-gmac-net.c:2982:87: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
 2982 |         cur_index, dma_desc->desc0, dma_desc->desc1, dma_desc->desc2, dma_desc->desc3);
      |                                                                                       ^
fuxi-gmac-net.c:2999:89: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
 2999 |         start_index, dma_desc->desc0, dma_desc->desc1, dma_desc->desc2, dma_desc->desc3);
      |                                                                                         ^
fuxi-gmac-net.c: In function 'fxgmac_dev_read':
fuxi-gmac-net.c:3116:94: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
 3116 |         if(netif_msg_rx_status(pdata)) DPRINTK("dev_read context desc,ch=%s\n",channel->name);
      |                                                                                              ^
fuxi-gmac-net.c:3194:86: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
 3194 |             DPRINTK("dev_read NOT last desc,pkt incomplete yet,%u\n", cnt_incomplete);
      |                                                                                      ^
fuxi-gmac-net.c:3199:86: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
 3199 |         DPRINTK("dev_read rx back to normal and incomplete cnt=%u\n", cnt_incomplete);
      |                                                                                      ^
  CC [M]  fuxi-gmac-pci.o
  CC [M]  fuxi-gmac-phy.o
fuxi-gmac-phy.c: In function 'fxgmac_phy_link_poll':
fuxi-gmac-phy.c:433:34: error: implicit declaration of function 'from_timer'; did you mean 'mod_timer'? [-Wimplicit-function-declaration]
  433 |     struct fxgmac_pdata *pdata = from_timer(pdata, t, expansion.phy_poll_tm);
      |                                  ^~~~~~~~~~
      |                                  mod_timer
fuxi-gmac-phy.c:433:55: error: 'expansion' undeclared (first use in this function)
  433 |     struct fxgmac_pdata *pdata = from_timer(pdata, t, expansion.phy_poll_tm);
      |                                                       ^~~~~~~~~
fuxi-gmac-phy.c:433:55: note: each undeclared identifier is reported only once for each function it appears in
fuxi-gmac-phy.c: In function 'fxgmac_phy_timer_init':
fuxi-gmac-phy.c:476:5: error: implicit declaration of function 'init_timer_key' [-Wimplicit-function-declaration]
  476 |     init_timer_key(&pdata->expansion.phy_poll_tm, NULL, 0, "fuxi_phy_link_update_timer", NULL);
      |     ^~~~~~~~~~~~~~
make[5]: *** [/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/source/scripts/Makefile.build:287: fuxi-gmac-phy.o] Error 1
make[4]: *** [/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/source/Makefile:2003: .] Error 2
make[3]: *** [/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/source/Makefile:248: __sub-make] Error 2
make[3]: Leaving directory '/build/source/src'
make[2]: *** [/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/source/Makefile:248: __sub-make] Error 2
make[2]: Leaving directory '/nix/store/c3m0i4hcqglfxw9c5axjjijrj1brqylx-linux-6.16-dev/lib/modules/6.16.0/build'
make[1]: *** [Makefile:32: modules] Error 2
make[1]: Leaving directory '/build/source/src'
make: *** [Makefile:33: modules] Error 2
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.15.8, NixOS, 25.11 (Xantusia), 25.11.20250725.6027c30`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.4`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/h15y13p2w17dhpiyh8pk42v1k4c38a0h-source`


### Notify maintainers

@5aaee9

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/os-specific/linux/yt6801/default.nix
+++ b/pkgs/os-specific/linux/yt6801/default.nix
@@ -12,7 +12,7 @@ let
 in
 stdenv.mkDerivation (finalAttrs: {
   pname = "yt6801";
-  version = "1.0.29-20240812";
+  version = "1.0.30-20250430";
 
   src =
     let
@@ -23,7 +23,7 @@ stdenv.mkDerivation (finalAttrs: {
     fetchzip {
       stripRoot = false;
       url = "https://www.motor-comm.com/Public/Uploads/uploadfile/files/${uploadDate}/yt6801-linux-driver-${versionName}.zip";
-      sha256 = "sha256-oz6CeOUN6QWKXxe3WUZljhGDTFArsknjzBuQ4IchGeU=";
+      sha256 = "sha256-6HeU3bbTaKOCy3X+nMpC9/bBc+0c4Ip5TdG+LGUGTKk=";
     };
 
   nativeBuildInputs = kernel.moduleBuildDependencies ++ [ kmod ];

--- a/pkgs/os-specific/linux/yt6801/default.nix
+++ b/pkgs/os-specific/linux/yt6801/default.nix
@@ -28,9 +28,13 @@ stdenv.mkDerivation (finalAttrs: {
 
   nativeBuildInputs = kernel.moduleBuildDependencies ++ [ kmod ];
 
-  patches = lib.optionals (lib.versionAtLeast kernel.version "6.15") [
-    ./kernel_6.15_fix.patch
-  ];
+  patches =
+    lib.optionals (lib.versionAtLeast kernel.version "6.15") [
+      ./kernel_6.15_fix.patch
+    ]
+    ++ lib.optionals (lib.versionAtLeast kernel.version "6.16") [
+      ./kernel_6_16_fix.patch
+    ];
 
   postPatch = ''
     substituteInPlace src/Makefile \

--- a/pkgs/os-specific/linux/yt6801/kernel_6_16_fix.patch
+++ b/pkgs/os-specific/linux/yt6801/kernel_6_16_fix.patch
@@ -1,0 +1,33 @@
+--- a/src/fuxi-gmac-phy.c	2025-08-03 00:00:00.000000000 +0000
++++ b/src/fuxi-gmac-phy.c	2025-08-03 00:00:00.000000000 +0000
+@@ -322,8 +322,10 @@
+ static void fxgmac_phy_link_poll(unsigned long data)
+ #endif
+ {
+-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,15,0))
++#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6,16,0))
++    struct fxgmac_pdata *pdata = container_of(t, struct fxgmac_pdata, expansion.phy_poll_tm);
++#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(4,15,0))
+     struct fxgmac_pdata *pdata = from_timer(pdata, t, expansion.phy_poll_tm);
+ #else
+     struct fxgmac_pdata *pdata = (struct fxgmac_pdata*)data;
+ #endif
+@@ -350,13 +352,17 @@
+
+ int fxgmac_phy_timer_init(struct fxgmac_pdata *pdata)
+ {
+-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,15,0))
++#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6,16,0))
++    timer_setup(&pdata->expansion.phy_poll_tm, fxgmac_phy_link_poll, 0);
++#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(4,15,0))
+     init_timer_key(&pdata->expansion.phy_poll_tm, NULL, 0, "fuxi_phy_link_update_timer", NULL);
+ #else
+     init_timer_key(&pdata->expansion.phy_poll_tm, 0, "fuxi_phy_link_update_timer", NULL);
+ #endif
+     pdata->expansion.phy_poll_tm.expires = jiffies + HZ / 2;
++#if (LINUX_VERSION_CODE < KERNEL_VERSION(6,16,0))
+     pdata->expansion.phy_poll_tm.function = (void *)(fxgmac_phy_link_poll);
++#endif
+ #if (LINUX_VERSION_CODE < KERNEL_VERSION(4,15,0))
+     pdata->expansion.phy_poll_tm.data = (unsigned long)pdata;
+ #endif

